lib/rouge/lexers/gdscript.rb

```diff
@@ -0,0 +1,201 @@
+          groups Text, Str::Affix, Str::Doc
+        end
+        rule %r/^(\s*)([rRuUbB]{,2})('''(?:.|\n)*?''')/ do
+          groups Text, Str::Affix, Str::Doc
+        end
+        rule %r/[^\S\n]+/, Text
+        rule %r/#.*$/, Comment::Single
+        rule %r/[\[\]{}:(),;]/, Punctuation
+        rule %r/\\\n/, Text
+        rule %r/\\/, Text
+        rule %r/(in|and|or|not)\b/, Operator::Word
+        rule %r/!=|==|<<|>>|&&|\+=|-=|\*=|\/=|%=|&=|\|=|\|\||[-~+\/*%=<>&^.!|$]/, Operator
+        rule %r/(func)((?:\s|\\\s)+)/ do
+          groups Keyword, Text
+          push :funcname
+        end
+        rule %r/(class)((?:\s|\\\s)+)/ do
+          groups Keyword, Text
+          push :classname
+        end
+        mixin :keywords
+        mixin :buildins
+        rule %r/(#{affix_long})(""")/ do
+          groups Str::Affix, Str::Double
+          push :tdqs
+        end
+        rule %r/(#{affix_long})(''')/ do
+          groups Str::Affix, Str::Double
+          push :tsqs
+        end
+        rule %r/(#{affix_long})(")/ do
+          groups Str::Affix, Str::Double
+          push :dqs
+        end
+        rule %r/(#{affix_long})(')/ do
+          groups Str::Affix, Str::Double
+          push :sqs
+        end
+        rule %r/(#{affix_short})(""")/ do
+          groups Str::Affix, Str::Double
+          push :escape_tdqs
+        end
+        rule %r/(#{affix_short})(''')/ do
+          groups Str::Affix, Str::Double
+          push :escape_tsqs
+        end
+        rule %r/(#{affix_short})(")/ do
+          groups Str::Affix, Str::Double
+          push :escape_dqs
+        end
+        rule %r/(#{affix_short})(')/ do
+          groups Str::Affix, Str::Double
+          push :escape_sqs
+        end
+        mixin :name
+        mixin :numbers
+      end
+
+      state :keywords do
+        rule %r/(#{keywords})\b/, Keyword
+        rule %r/(?<!\.)(#{keywords_reserved})\b/, Keyword::Reserved
+      end
+
+      state :buildins do
+        rule %r/(?<!\.)(#{buildins})\b/, Name::Builtin
+        rule %r/((?<!\.)(self|false|true)|(PI|TAU|NAN|INF))\b/, Name::Builtin::Pseudo
+        rule %r/(?<!\.)(#{builtins_type})\b/, Keyword::Type
+      end
+
+      state :numbers do
+        rule %r/(\d+\.\d*|\d*\.\d+)([eE][+-]?[0-9]+)?j?/, Num::Float
+        rule %r/\d+[eE][+-]?[0-9]+j?/, Num::Float
+        rule %r/0[xX][a-fA-F0-9]+/, Num::Hex
+        rule %r/\d+j?/, Num::Integer
+      end
+
+      state :name do
+        rule %r/[a-zA-Z_]\w*/, Name
+      end
+
+      state :funcname do
+        rule %r/[a-zA-Z_]\w*/, Name::Function, :pop!
+      end
+
+      state :classname do
+        rule %r/[a-zA-Z_]\w*/, Name::Class, :pop!
+      end
+
+      state :string_escape do
+        rule %r/\\([\\abfnrtv"\']|\n|N\{.*?\}|u[a-fA-F0-9]{4}|U[a-fA-F0-9]{8}|x[a-fA-F0-9]{2}|[0-7]{1,3})/, Str::Escape
+      end
+
+      state :strings_single do
+        rule %r/%(\(\w+\))?[-#0 +]*([0-9]+|[*])?(\.([0-9]+|[*]))?[hlL]?[E-GXc-giorsux%]/, Str::Interpol
```

Comment:
How does string interpolation work in GDScript?

Reply:
Similarly to Python: [Godot Docs](https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_format_string.html)